src/iocore/net/unit_tests/sni_conf_test.yaml

```diff
@@ -36,3 +36,16 @@ sni:
   http2_buffer_water_mark: 256
 - fqdn: foo.bar.com
   http2: false
+
+# test with mixed-case
+- fqdn: "MiXeDcAsE.foo.com"
+  http2: true
+  http2_buffer_water_mark: 256
+  inbound_port_ranges: 31337
+
+# test with mixed-case glob
+- fqdn: "*.MiXeDcAsE.com"
+  http2: false
+
+# test glob in the middle, this will be an exact match
+- fqdn: "cat.*.com"
```

Comment:
Discussion around the #9736, we concluded don't allow this. This should be error.

> 1. Allow single left-most `*` in the fqdn field

https://docs.trafficserver.apache.org/en/latest/admin-guide/files/sni.yaml.en.html